### PR TITLE
Witness api plugin implemented #618

### PR DESCRIPTION
**Pull request by vpavliv**
Moved methods from database_api to a new plugin

### CODE REVIEW

File: plugins/database_api/api.cpp [hunk 1]
```diff
@@ -78,11 +81,7 @@ struct plugin::api_impl final {
     std::set<std::string> lookup_accounts(const std::string &lower_bound_name, uint32_t limit) const;
     uint64_t get_account_count() const;
 
-    // Witnesses
-    std::vector<optional<witness_api_object>> get_witnesses(const std::vector<witness_object::id_type> &witness_ids) const;
-    fc::optional<witness_api_object> get_witness_by_account(std::string account_name) const;
-    std::set<account_name_type> lookup_witness_accounts(const std::string &lower_bound_name, uint32_t limit) const;
-    uint64_t get_witness_count() const;
+    std::vector<optional<witness_api::witness_api_object>> get_witnesses(const std::vector<witness_object::id_type> &witness_ids) const;
```
Comment (afalaleev) [inline plugins/database_api/api.cpp]: What reasons to keep this method?

File: plugins/database_api/api.cpp [hunk 2]
```diff
@@ -22,6 +23,8 @@
 
 namespace golos { namespace plugins { namespace database_api {
 
+    using namespace golos::plugins::witness_api;
```
Comment (afalaleev) [inline plugins/database_api/api.cpp]: For which cases are used this namespace?

File: plugins/database_api/include/golos/plugins/database_api/api_objects/owner_authority_history_api_object.hpp [hunk 3]
```diff
@@ -7,6 +7,8 @@ namespace golos {
     namespace plugins {
         namespace database_api {
 
+            using protocol::authority;
+            using golos::protocol::account_name_type;
```
Comment (afalaleev) [inline plugins/database_api/include/golos/plugins/database_api/api_objects/owner_authority_history_api_object.hpp]: Why did you add these lines?

File: plugins/database_api/include/golos/plugins/database_api/state.hpp [hunk 4]
```diff
@@ -4,14 +4,14 @@
...
 #include <golos/chain/steem_object_types.hpp>
-#include <golos/plugins/database_api/api_objects/witness_api_object.hpp>
 #include "forward.hpp"
 #include <golos/api/extended_account.hpp>
 
 namespace golos {
     namespace plugins {
         namespace database_api {
 
+            using golos::protocol::asset;
```
Comment (afalaleev) [inline plugins/database_api/include/golos/plugins/database_api/state.hpp]: For what?

File: plugins/database_api/api.cpp (hunk 1 above)
Comment (vpavliv) [reply plugins/database_api/api.cpp]: It is used inside database_api so i removed it from database_api plugin but left it for database needs

File: plugins/database_api/api.cpp (hunk 2 above)
Comment (vpavliv) [reply plugins/database_api/api.cpp]: It is needed to resolve witness_api_object from get_witnesses. i moved witness_api_object to other witness_api. But i guess we dont need whole namespace here, its probably beter to just resolve witness_api_object in places.

File: plugins/database_api/include/golos/plugins/database_api/api_objects/owner_authority_history_api_object.hpp (hunk 3 above)
Comment (vpavliv) [reply plugins/database_api/include/golos/plugins/database_api/api_objects/owner_authority_history_api_object.hpp]: I moves witness_api_object and feed_history_api_object to other directory, and those files had plenty of using's so after i had to restore some of those to make project buildable

File: plugins/database_api/include/golos/plugins/database_api/state.hpp (hunk 4 above)
Comment (vpavliv) [reply plugins/database_api/include/golos/plugins/database_api/state.hpp]: this using was in witness_api_object file that i have moved to other plugin so i needed to restore some of using's

File: plugins/database_api/api.cpp (hunk 1 above)
Comment (vpavliv) [reply plugins/database_api/api.cpp]: Yes my bad i will remove it.

File: plugins/database_api/api.cpp (hunk 2 above)
Comment (vpavliv) [reply plugins/database_api/api.cpp]: Yes i will remove it

File: plugins/witness_api/include/golos/plugins/witness_api/api_objects/feed_history_api_object.hpp [hunk 5]
```diff
@@ -1,14 +1,15 @@
-#ifndef GOLOS_FEED_HISTORY_API_OBJ_HPP
-#define GOLOS_FEED_HISTORY_API_OBJ_HPP
+#ifndef GOLOS_FEED_HISTORY_WITNESS_API_OBJ_HPP
+#define GOLOS_FEED_HISTORY_WITNESS_API_OBJ_HPP
```
[the remark this answers is not in the document]
Comment (afalaleev) [reply plugins/witness_api/include/golos/plugins/witness_api/api_objects/feed_history_api_object.hpp]: Yes, I agree. It make header files more simple.

[the remark this answers is not in the document]
Comment (vpavliv) [reply plugins/witness_api/include/golos/plugins/witness_api/api_objects/feed_history_api_object.hpp]: Ok. Done.
